classes/class-uagb-post-assets.php

```diff
@@ -385,11 +385,16 @@ public function allow_assets_generation() {
 	 * @since 1.23.0
 	 */
 	public function enqueue_scripts() {
-		global $_wp_current_template_content;
-		$blocks = parse_blocks( $_wp_current_template_content );
-		// Global Required assets.
-		// If the current template has content and contains blocks, execute this code block.
-		if ( has_blocks( $this->post_id ) || ( $_wp_current_template_content && has_blocks( $blocks ) ) ) {
+			$blocks                              = array();
+			$spectra_wp_current_template_content = '';
+		if ( function_exists( 'wp_is_block_theme' ) && wp_is_block_theme() ) {
+			global $_wp_current_template_content;
+			$spectra_wp_current_template_content = $_wp_current_template_content;
+			$blocks                              = parse_blocks( $spectra_wp_current_template_content );
+		}
+			// Global Required assets.
+			// If the current template has content and contains blocks, execute this code block.
+		if ( has_blocks( $this->post_id ) || ( $spectra_wp_current_template_content && has_blocks( $blocks ) ) ) {
```

Comment:
```suggestion
		if ( has_blocks( $this->post_id ) || has_blocks( $blocks ) ) {
```
@AbhijitNage123 do we need to check for $spectra_wp_current_template_content as we are getting blocks from it only.
- alos i am curious if there can be a case thta current theme is block theme but $_wp_current_template_content is still null.

Reply:
@adi3890 In block theme, $_wp_current_template_content never return null, it always has an value in block theme.

Reply:
Resolved.